Mathlib/AlgebraicGeometry/EllipticCurve/Projective.lean

```diff
@@ -41,12 +42,12 @@ given by a tuple consisting of $[x:y:z]$ and the nonsingular condition on any re
 ## Implementation notes
 
 A point representative is implemented as a term `P` of type `Fin 3 → R`, which allows for the vector
-notation `![x, y, z]`. However, `P` is not definitionally equivalent to the expanded vector
-`![P x, P y, P z]`, so the auxiliary lemma `fin3_def` can be used to convert between the two forms.
-The equivalence of two point representatives `P` and `Q` is implemented as an equivalence of orbits
-of the action of `Rˣ`, or equivalently that there is some unit `u` of `R` such that `P = u • Q`.
-However, `u • Q` is again not definitionally equal to `![u * Q x, u * Q y, u * Q z]`, so the
-auxiliary lemmas `smul_fin3` and `smul_fin3_ext` can be used to convert between the two forms.
+notation `![x, y, z]`. However, `P` is not syntactically equivalent to the expanded vector
+`![P x, P y, P z]`, so the lemmas `fin3_def` and `fin3_def_ext` can be used to convert between the
+two forms. The equivalence of two point representatives `P` and `Q` is implemented as an equivalence
+of orbits of the action of `Rˣ`, or equivalently that there is some unit `u` of `R` such that
+`P = u • Q`. However, `u • Q` is not syntactically equal to `![u * Q x, u * Q y, u * Q z]`, so the
+lemmas `smul_fin3` and `smul_fin3_ext` can be used to convert between the two forms.
```

Comment:
Sure, but not being syntactically equal implies not being definitionally equal, so this is still technically correct. I think `u • Q` is definitionally equal to `![u * Q x, u * Q y, u * Q z]` (since the proof is `rfl`) but not syntactically equal.

I'm mirroring https://github.com/leanprover-community/mathlib4/pull/13060 so I'd have to change both in this PR if you prefer.

Reply:
The proof is `List.ofFn_inj.mp rfl` not rfl. It's different from the Jacobian case since the scalar multiplication here is the general one for any function, not necessarily a 3-tuple.